pkg/kv/kvserver/replica_consistency.go

```diff
@@ -70,8 +70,9 @@ var fatalOnStatsMismatch = envutil.EnvOrDefaultBool("COCKROACH_ENFORCE_CONSISTEN
 // replicaChecksum contains progress on a replica checksum computation.
 type replicaChecksum struct {
 	CollectChecksumResponse
-	// started is true if the checksum computation has started.
-	started bool
+	// stop terminates the checksum computation. It is not nil iff the computation
+	// has started.
+	stop func()
```

Comment:
This is created by the computation task, so we can only cancel it from the request if the computation has started. We should also make sure that if the request exited, it marks this `replicaChecksum` in such a way that the computation won't start.

For example, we can store `Context+cancel` here instead, created by whoever first creates this struct. The `Context` will be used by the task, and `cancel` can kill it even before the task has started.

Reply:
Added a `gaveUp` field to address this case.